compiler/tests-common/tests/org/jetbrains/kotlin/codegen/AbstractSteppingTest.kt

```diff
@@ -0,0 +1,202 @@
+                        finish = true
+                    }
+                    is VMStartEvent -> {
+                        val classReq = manager.createClassPrepareRequest()
+                        classFileFactory.getClassFiles().map { classReq.addClassFilter(it.internalName) }
+                        classReq.setSuspendPolicy(SUSPEND_ALL)
+                        classReq.enable()
+                    }
+                    is ClassPrepareEvent -> {
+                        val methodEntryReq = manager.createMethodEntryRequest()
+                        classFileFactory.getClassFiles().map { methodEntryReq.addClassFilter(it.internalName) }
+                        methodEntryReq.setSuspendPolicy(SUSPEND_ALL)
+                        methodEntryReq.enable()
+
+                        val methodExitReq = manager.createMethodExitRequest()
+                        classFileFactory.getClassFiles().map { methodExitReq.addClassFilter(it.internalName) }
+                        methodExitReq.setSuspendPolicy(SUSPEND_ALL)
+                        methodExitReq.enable()
+
+                        val stepReq = manager.createStepRequest(event.thread(), StepRequest.STEP_LINE, StepRequest.STEP_INTO)
+                        stepReq.setSuspendPolicy(SUSPEND_ALL)
+                        classFileFactory.getClassFiles().map { stepReq.addClassFilter(it.internalName) }
+                        stepReq.enable()
+                    }
+                    is MethodEntryEvent -> {
+                        if (event.location().method().name() == "box") {
+                            inBox = true
+                        }
+                    }
+                    is StepEvent -> {
+                        if (inBox) {
+                            loggedEvents.add(event)
+                        }
+                    }
+                    is MethodExitEvent -> {
+                        if (event.location().method().name() == "box") {
+                            finish = true
+                        }
+                    }
+                    else -> {
+                    }
+                }
+            }
+            if (finish) {
+                break
+            }
+            eventSet.resume()
+        }
+        val actual = loggedEvents.map { (it as StepEvent).location().lineNumber() }
+        TestCase.assertEquals(lineNumbers, actual)
+    }
+
+    private fun isPortOpen(): Boolean {
+        return try {
+            Socket(DEBUG_ADDRESS, DEBUG_PORT).close()
+            true
+        } catch (e: IOException) {
+            false
+        }
+    }
+
+    private fun startDebuggeeProcess(classesDir: File): Process {
+        val classesToLoad = this.classFileFactory.getClassFiles()
+            .map { it.qualifiedName }
+            .joinToString(",")
+
+        val classpath = listOf(
+            classesDir.absolutePath,
+            PathUtil.getJarPathForClass(Delegates::class.java) // Add Kotlin runtime JAR
```

Comment:
Is this needed currently for the tests?

Reply:
Mainly for extending to multi-file test cases.